backend/src/main.rs

```diff
@@ -0,0 +1,74 @@
+#![allow(clippy::multiple_crate_versions)]
+use std::path::PathBuf;
+
+use axum::{routing::get, Router};
+use backend::{prelude::*, root, services::api_router};
+use cli::Parser;
+use tower::ServiceBuilder;
+use tower_http::cors::CorsLayer;
+use tracing::Level;
+use utoipa::OpenApi;
+use utoipa_rapidoc::RapiDoc;
+use utoipa_redoc::{Redoc, Servable};
+use utoipa_swagger_ui::SwaggerUi;
+
+#[tokio::main]
+#[allow(clippy::unwrap_used, clippy::expect_used)]
+async fn main() -> Result<()> {
```

Comment:
How does it work, when you have both `lib.rs`  and `main.rs` in the crate?

Reply:
All the core backend features will be located in the `lib.rs`, while the actual start of the backend server will be in `main.rs` file. In such a way, there will be both library and binary with the same name. The library can be referenced from the binary via `use backend::...` .
Why do we care?
With such approach, we can import the codebase as dependency and swap out `main.rs` to some other binary (the example of this is [gen_openapi.rs](https://github.com/qoollo/bob-management/pull/13/files#diff-21dd2769c434ea847ba4341448dfbb34a999f361cf15444ca71e28f573bc45a9) file in #13). There we define ApiDoc in `lib.rs` file, that is responsible for API documentation of the project, and use it to generate corresponding files of that documentation